src/components/MovieTile/MovieTile.tsx

```diff
@@ -0,0 +1,55 @@
+import './MovieTile.css';
+import MovieTileModel from '../../models/MovieTileModel';
+import DotsIcon from './../../resources/DotsIcon.png';
+import { useState } from 'react';
+
+interface MovieTileProps {
+  movieModel: MovieTileModel;
+  onClickCallback: (movieName: string) => void;
+  onEditClickCallback: (movieName: string) => void;
+  onDeleteClickCallback: (movieName: string) => void;
+}
+
+function MovieTile({ movieModel, onClickCallback, onEditClickCallback, onDeleteClickCallback }: MovieTileProps) {
+    const [isContextMenuOpen, setContextMenuOpen] = useState(false);
+
+    const toggleContextMenu = () => {
+        setContextMenuOpen(!isContextMenuOpen);
```

Comment:
It's better to rewrite it as (useState is asynchronous, to avoid unexpected value in isContextMenuOpen): 
` setContextMenuOpen(prevValue => !prevValue);`